rskj-core/src/main/java/co/rsk/net/handler/TxPendingValidator.java

```diff
@@ -65,14 +66,14 @@ public TxPendingValidator(Constants constants, ActivationConfig activationConfig
         validatorSteps.add(new TxValidatorMaximumGasPriceValidator(activationConfig));
     }
 
-    public TransactionValidationResult isValid(Transaction tx, Block executionBlock, @Nullable AccountState state) {
+    public TransactionValidationResult isValid(Transaction tx, Block executionBlock, @Nullable AccountState state, RskAddress sender) {
         BigInteger blockGasLimit = BigIntegers.fromUnsignedByteArray(executionBlock.getGasLimit());
         Coin minimumGasPrice = executionBlock.getMinimumGasPrice();
         long bestBlockNumber = executionBlock.getNumber();
         long basicTxCost = tx.transactionCost(constants, activationConfig.forBlock(bestBlockNumber));
 
         if (state == null && basicTxCost != 0) {
-            logger.trace("[tx={}, sender={}] account doesn't exist", tx.getHash(), tx.getSender());
+            logger.trace("[tx={}, sender={}] account doesn't exist", tx.getHash(), sender);
```

Comment:
just an idea.. what about providing signature cache in the ctor and using it here? it's not that obvious that new param - sender - has to be sender of the tx being provided..

Reply:
It sounds good! if it's clearer let's do it! I thought that maybe it was too much to include this in the ctor just for a log, but you're right, it will look better. Will do.

Reply:
done!